lib/src/client/http2_connection.dart

```diff
@@ -78,7 +78,7 @@ class Http2ClientConnection implements connection.ClientConnection {
   Future<Socket> _createSocket() async {
     final securityContext = credentials.securityContext;
     if (securityContext == null) {
-      return Socket.connect(host, port);
+      return Socket.connect(host, port, timeout: options.connectTimeout);
```

Comment:
There are two more places a socket is created in the same function. They need to be updated as well.